File: Seccion 11 - Funciones en C++/Seccion_11_Dart/28_ejercicio_22.dart

import 'dart:io';

int escribeNumeros(int ini, int fin) {
  if (ini == fin) {
    return ini;
  } else {
    return escribeNumeros(ini, fin - 1);
  }
}

void main() {

  int ini, fin;

  stdout.writeln('Digite un inicio: ');
  ini = int.parse(stdin.readLineSync());
  stdout.writeln('Digite un final: ');
  fin = int.parse(stdin.readLineSync());

  for (int i = ini; i <= fin; i++) {
    stdout.write('${escribeNumeros(i, fin)}, ');
  }

  print('');
}

File: Seccion 11 - Funciones en C++/Seccion_11_Python/28_ejercicio_22.py

import typing

def escribe_numeros(ini: int, fin: int):
    if ini == fin:
        return ini
    else:
        return escribe_numeros(ini, fin - 1)

def main():
    
    ini: int = 0
    fin: int = 0
    
    ini = int(input('Digite un inicio: '))
    fin = int(input('Digite un final: '))
    
    for i in range(ini, fin + 1):
        print('{}, '.format(escribe_numeros(i, fin)), end='')
    
    print()

if __name__ == '__main__':
    main()